textacy/viz/termite.py

```diff
@@ -200,3 +209,131 @@ def draw_termite_plot(
+            topics (int or Sequence[int]): topic(s) to include in termite plot;
+                if -1, all topics are included
+            highlight_topics (str or Sequence[str]): names for up to 6 topics
+                to visually highlight in the plot with contrasting colors
+            n_terms (int): number of top terms to include in termite plot
+            rank_terms_by ({'max', 'mean', 'var'}): argument to dataframe `agg`
+                function, used to rank terms;
+                the top-ranked ``n_terms`` are included in the plot
+            sort_terms_by ({'seriation', 'weight', 'index', 'alphabetical'}):
+                method used to vertically sort the selected top ``n_terms`` terms;
+                the default ("seriation") groups similar terms together, which
+                facilitates cross-topic assessment
+            save (str): give the full /path/to/fname on disk to save figure
+                rc_params (dict, optional): allow passing parameters to rc_context
+                in matplotlib.plyplot, details in
+                https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.rc_context.html
+        Returns:
+            ``matplotlib.axes.Axes.axis``: Axis on which termite plot is plotted.
+        Raises:
+            ValueError: if more than 6 topics are selected for highlighting, or
+                an invalid value is passed for the sort_topics_by, rank_terms_by,
+                and/or sort_terms_by params
+        References:
+            - Chuang, Jason, Christopher D. Manning, and Jeffrey Heer. "Termite:
+              Visualization techniques for assessing textual topic models."
+              Proceedings of the International Working Conference on Advanced
+              Visual Interfaces. ACM, 2012.
+            - Fajwel Fogel, Alexandre d’Aspremont, and Milan Vojnovic. 2016.
+              Spectral ranking using seriation. J. Mach. Learn. Res. 17, 1
+              (January 2016), 3013–3057.
+        See Also:
+            :func:`viz.termite_plot <textacy.viz.termite.termite_plot>`
+        TODO: `rank_terms_by` other metrics, e.g. topic salience or relevance
+        """
+        if highlight_topics is not None:
+            if isinstance(highlight_topics, str):
+                highlight_topics = [highlight_topics]
+            elif len(highlight_topics) > 6:
+                raise ValueError("no more than 6 topics may be highlighted at once")
+
+        # get column index of any topics to highlight in termite plot
+        if highlight_topics is not None:
+            highlight_cols = tuple(
+                components.columns.get_loc(c)
+                for c in highlight_topics
+            )
+        else:
+            highlight_cols = None
+
+        component_filter = (components.loc[
+            components
+            .agg(rank_terms_by, axis=1)
+            .sort_values(ascending=False)
+            .iloc[:n_terms]
+            .index
+        ])
```

Comment:
does this work if `components` is a sparse matrix, as described in the docstring?

Reply:
AH I had meant to write a check to see if a sparse mat was passed, but didn't want to reverse engineer the expected matrices from your function. 

I highly recommend interfacing with plots by exploiting pandas. To be clear, pandas sparse dtypes _will work_. And if a csr mat is passed (and we check for shape compatibility) we can load it up into a dataframe automatically. 

Ill see if some of the tests I write can cover the bases for columnar dtypes (see the new String ant Sparse types pandas provides in their docs).

Reply:
- [string dtype](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.0.0.html#dedicated-string-data-type)
- [Others, incl sparse](https://pandas.pydata.org/pandas-docs/stable/getting_started/basics.html#basics-dtypes)